tests/tests_restore_dirs.yml

```
@@ -0,0 +1,54 @@
+---
+- name: Check if selinux role restores directory SELinux context
+  hosts: all
+  tasks:
+    - name: Create tempdir, set fcontext and relabel
+      block:
+        - name: Create temporary directory
+          ansible.builtin.tempfile:
+            state: directory
+            prefix: linux_system_role.selinux
+          register: tempdir
+
+        - name: Create temporary file
+          ansible.builtin.tempfile:
+            path: "{{ tempdir.path }}"
+            state: file
+            suffix: temp
+          register: tempfile
+
+        - name: Set default_t context to {{ tempdir.path }}
+          ansible.builtin.command: chcon -R -t default_t "{{ tempdir.path }}"
+          changed_when: true
+
+        - name: Run the role
+          include_role:
+            name: linux-system-roles.selinux
+          vars:
+            selinux_fcontexts:
+              - {target: "{{ tempdir.path }}(/.*)?", setype: 'user_tmp_t'}
+            selinux_restore_dirs:
+              - "{{ tempdir.path }}"
+
+        - name: Get SELinux context of {{ tempfile.path }}
+          ansible.builtin.command: ls -Z {{ tempfile.path }}
```

```suggestion
          command: ls -Z {{ tempfile.path | quote }}
```
use `quote` here - or not - not sure if you need to quote the use of `tempfile.path` - but if you do, please be consistent

---

fixed